### PR TITLE
base/{sim,steer}: override, const, cleanup

### PR DESCRIPTION
**Pull request by ChristianTackeGSI**
* FairVoume: Major cleanup
* Rest: override

---

Checklist:

* [X] Rebased against `dev` branch
* [X] My name is in the resp. CONTRIBUTORS/AUTHORS file
* [X] Followed [the seven rules of great commit messages](https://chris.beams.io/posts/git-commit/#seven-rules)

### CODE REVIEW

File: base/sim/FairVolume.cxx [hunk 1]
```diff
@@ -14,56 +14,18 @@
...
-    , /**Volume Copy No*/
-    fMotherId(-1)
-    , /**Mother Volume Id*/
-    fMotherCopyNo(-1)
-    , /**Mother Volume Copy No*/
-    fDetector(0)
-    , fModule(nullptr)
-    ,              /**The Module (detector) which will proccess the hits for this volume*/
-    fNode(nullptr) /**Node corre*/
+    , fMotherId(-1)
```
[the remark this answers is not in the document]
Comment (fuhlig1) [reply base/sim/FairVolume.cxx:17]: I am not 100% sure but -1 should be okay for both constructors. I will try to have a deeper look tomorrow.

File: base/sim/FairVolume.h [hunk 2]
```diff
@@ -27,57 +27,51 @@ class FairVolume : public TNamed
 {
   public:
     FairVolume();
-    FairVolume(TString name, Int_t id = 0, Int_t detid = 0, FairModule* fMod = 0);
-    virtual ~FairVolume();
-
-    //    const char*  GetName() { return fName.Data();}
-    //    TString getName() { return fName;}
+    FairVolume(TString name, Int_t id = 0, Int_t ModId = 0, FairModule* fMod = nullptr);
```
Comment (ChristianTackeGSI) [inline base/sim/FairVolume.h:30]: @fuhlig1 Can you please also comment on `fModId`?

The default `()` constructor sets it to `-1` (that's why I have set the member initializer to that value), but the default argument for this constructor also has a `0`?

File: base/sim/FairVolume.cxx (hunk 1 above)
[the remark this answers is not in the document]
Comment (fuhlig1) [reply base/sim/FairVolume.cxx:19]: I checked and couldn' t find any place where the the value is checked to be 0 or -1.

[the remark this answers is not in the document]
Comment (fuhlig1) [reply base/sim/FairVolume.cxx:19]: Can be put in the header file as -1.

**Review by fuhlig1 — CHANGES_REQUESTED** [review]
Can be merged after fMotherID is initialized in the header file.

**Comment by ChristianTackeGSI** [comment]
Okay, added the member initialiser for `fMotherID{-1}` and rebased.

**Review by fuhlig1 — APPROVED** [review]

**Pull request merged**